swift_scality_backend/policy_configuration.py

```diff
@@ -0,0 +1,612 @@
+
+        :raise ValueError: Duplicate policy index found
+        '''
+
+        self._policies_map = {}
+
+        for policy in policies:
+            if policy.index in self._policies_map:
+                raise ValueError('Duplicate policy index: %r' % policy.index)
+
+            self._policies_map[policy.index] = policy
+
+    def __repr__(self):
+        return 'Configuration(policies=%r)' % list(self)
+
+    def __eq__(self, other):
+        if not isinstance(other, Configuration):
+            return NotImplemented
+
+        return frozenset(self) == frozenset(other)
+
+    def __ne__(self, other):
+        equal = self.__eq__(other)
+
+        if equal is NotImplemented:
+            return NotImplemented
+        else:
+            return not equal
+
+    def __iter__(self):
+        return self._policies_map.itervalues()
+
+    def __hash__(self):
+        return hash(frozenset(self))
+
+    def get_policy(self, index):
+        '''Retrieve a policy based on its index
+
+        :param index: Policy to look up
+        :type index: Type of `StoragePolicy.index`
+
+        :raise ValueError: Unknown policy index
+        '''
+
+        if index in self._policies_map:
+            return self._policies_map[index]
+        else:
+            raise ValueError('Unknown policy index: %r' % index)
+
+    @classmethod
+    def from_stream(cls, stream, filename=None):
+        '''Parse a configuration from a stream
+
+        This functions turns a configuration file into a `Configuration`, with
+        some sanity checks along the way. It is based on
+        `ConfigParser.SafeConfigParser`.
+
+        The `stream` object must have a `readline` method. `filename` will be
+        used in error reporting, if available.
+
+        :param stream: Stream to parse
+        :type stream: File-like object
+        :param filename: Filename of input
+        :type filename: `str`
+
+        :returns: Parsed `Configuration`
+        :rtype: `Configuration`
+
+        :raise ConfigurationError: Various configurations issues detected
+        '''
+
+        parser = ConfigParser.SafeConfigParser()
+        parser.readfp(stream, filename)
+
+        ring_sections = [
+            section for section in parser.sections()
+            if section.startswith(RING_SECTION_PREFIX)]
+        sp_sections = [
+            section for section in parser.sections()
+            if section.startswith(STORAGE_POLICY_SECTION_PREFIX)]
+
+        _default = object()
+
+        def get(section, setting, default=_default):
+            '''Safely retrieve a value from a configuration.'''
+
+            if not parser.has_option(section, setting):
+                if default is not _default:
+                    return default
+                else:
+                    raise ConfigurationError(
+                        'Section %r lacks a %r setting' % (section, setting))
+
+            return parser.get(section, setting)
+
+        def split_list(val):
```

Comment:
I need exactly this to parse the configuration of the swift-object-server, the sproxyd_host flag.

Could you move this function in utils.py